### ISSUE TITLE
[Tags] Discussions using nested secondary tags without primary after move

### ISSUE THREAD

**Issue by jordanjay29** [flarum/core#828]
On the recent discuss.flarum move, it was discovered that moving a primary tag X under another primary Y (so that X becomes a nested secondary tag under Y) will not auto-add the new primary Y tag to any discussions under X. This may be unexpected behavior (@tobscure weigh in?) as a user cannot start a new discussion with this setup (using nested X but not Y).

**Comment by tobscure** [comment]
Yes this is unexpected behaviour. Will keep it in mind when I overhaul the tags extension one day.

**Comment by Luceos** [comment]
Let's keep this issue as my experience with the refactor.

Some stuff in advance:

- Before starting I was aware of the possibility that moving a primary with subtags under another primary would probably result in those secondaries being lost in the overview. They would not be deleted, they would still be attached to the discussion, but they weren't visible in the tags overview anymore.

What happened:

- Moving a subtag into another primary would not update the tags of the discussions in any way. The old primaries were still registered on the discussions. For instance I moved showcase from resources to off-topic. When clicking off-topic you'll see the subtag and in that subtag the discussions, but the primary of those discussions is still resources, not off-topic.
- Moving a subtag to another primary and deleting the old primary as a consequence will result in discussions from the subtag only listed under that tag and not in any other primary.
